src/Disks/ObjectStorages/S3/diskSettings.cpp

```diff
@@ -142,6 +142,7 @@ std::unique_ptr<Aws::S3::S3Client> getClient(const Poco::Util::AbstractConfigura
         uri.is_virtual_hosted_style,
         config.getString(config_prefix + ".access_key_id", ""),
         config.getString(config_prefix + ".secret_access_key", ""),
+        {},
```

Comment:
Yeah, probably. I've thought there is no sense doing that since it is only a problem when credentials are temporary so that user can quickly copy-paste it to the query, and in config file they will become obsolete quite soon.